habitat-lab/habitat/gui/replay_gui_app_renderer.py

```diff
@@ -57,11 +66,27 @@ def post_sim_update(self, post_sim_update_dict):
         for keyframe in keyframes:
             self._replay_renderer.set_environment_keyframe(env_index, keyframe)
 
+        if len(keyframes):
+            self._need_render = True
+
         if "debug_images" in post_sim_update_dict:
             self._debug_images = post_sim_update_dict["debug_images"]
 
-        if len(keyframes):
-            self._need_render = True
+        if "info" in post_sim_update_dict:
+            # get needed metrics and format text string
+            # to be drawn on the GUI app screen
+            info = post_sim_update_dict["info"]
+            self._text_to_draw = (
+                pformat(
+                    {
+                        "App metrics": info["app"],
+                        "Env metrics": info["env"],
```

Comment:
This is a bit too specific to put into ReplayGuiAppRenderer. E.g. the assumption that we only want to print app and env metrics as opposed to arbitrary text.

How about just give the Driver direct access to the TextDrawer? Similar to how we give direct access to the Debug Line Render. See my thoughts below on the interface; we have to separate adding/drawing text from actually drawing to the screen.

Reply:
Done